src/ConfCrypt/Validation.hs

```diff
@@ -52,29 +52,29 @@ parameterTypesMatchSchema key ConfCryptFile {parameters} =
     where
         decryptAndCompare Parameter {paramName, paramValue, paramType} =
             catchError (runRule paramType paramName =<< decryptValue key paramValue)
-                       (const $ tell ["Error: Could not decrypt " <> paramName])
+                       (pure $ liftIO $ putStrLn ("Error: Could not decrypt " <> T.unpack paramName))
         runRule paramType paramName val =
             case paramType of
                 Nothing -> pure ()
                 Just CInt | all isDigit $ T.unpack val -> pure ()
                 Just CBoolean | T.toLower val == "true" || T.toLower val == "false" -> pure ()
                 Just CString | not (T.null val) -> pure ()
-                Just CString | T.null val -> tell ["Warning: "<> paramName <> " is empty"]
-                Just pt -> tell ["Error: "<> paramName <> " does not match the schema type " <> typeToOutputString pt]
+                Just CString | T.null val -> liftIO $ putStrLn ("Warning: "<> T.unpack paramName <> " is empty")
```

Comment:
Same here with accumulating the `Text` warning messages rather than printing them.

Reply:
Errors are now accumulated in last commit.